RSSDP/SsdpDevicePublisher.cs

```diff
@@ -427,7 +442,12 @@ private void SendAliveNotification(SsdpDevice device, string notificationType, s
 
             var message = BuildMessage(header, values);
 
-            _CommsServer.SendMulticastMessage(message, cancellationToken);
+            if (_sendOnlyMatchedHost)
+            {
+                _CommsServer.SendMulticastMessage(message, rootDevice.Address, cancellationToken);
```

Comment:
this is exactly where a ternary could fit IMHO, look:
```suggestion
                _CommsServer.SendMulticastMessage(message, _sendOnlyMatchedHost ? rootDevice.Address : null, cancellationToken);
```